commit 725d0a2472b77194729560dc9235fc03ebb0d978
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2019-06-03T19:49:13+00:00 (4 months ago)
    
    joyent/mdb_v8#114 error: left shift of negative value
    Reviewed by: Colin Ihrig <colin.ihrig@joyent.com>
    Reviewed by: John Levon <john.levon@joyent.com>
    Approved by: John Levon <john.levon@joyent.com>

diff --git a/src/mdb_v8.c b/src/mdb_v8.c
index da901f5..45892da 100644
--- a/src/mdb_v8.c
+++ b/src/mdb_v8.c
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -3216,7 +3216,7 @@ jsfunc_lineno(uintptr_t lendsp, uintptr_t tokpos,
 		 * we must convert it here, unless we're checking against the
 		 * "-1" sentinel.
 		 */
-		if (tokpos == V8_VALUE_SMI(-1))
+		if (tokpos == V8_VALUE_SMI((uintptr_t)-1))
 			mdb_snprintf(buf, buflen, "unknown position");
 		else
 			mdb_snprintf(buf, buflen, "position %d",
diff --git a/src/mdb_v8_function.c b/src/mdb_v8_function.c
index 7d11284..8e34588 100644
--- a/src/mdb_v8_function.c
+++ b/src/mdb_v8_function.c
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -453,7 +453,7 @@ v8funcinfo_definition_location(v8funcinfo_t *fip, mdbv8_strbuf_t *strb,
 	 * into the script).
 	 */
 	if (fip->v8fi_line_endings == NULL) {
-		if (tokpos == V8_VALUE_SMI(-1)) {
+		if (tokpos == V8_VALUE_SMI((uintptr_t)-1)) {
 			mdbv8_strbuf_sprintf(strb, "unknown position");
 		} else {
 			/*
